src/Apps/Order/Components/AddressModal.tsx

```diff
@@ -104,6 +104,25 @@ export const AddressModal: React.FC<Props> = ({
     setCreateUpdateError(null)
   }
 
+  const getInitialFormikValues = () => {
+    const initialPhoneValues = {
+      isValid: false,
+      national: "",
+      regionCode: "",
+    }
+
+    return createMutation
+      ? { country: "US", phone: initialPhoneValues }
+      : {
+          ...address,
+          phone: {
+            isValid: address?.phoneNumber && address?.country,
+            national: address?.phoneNumber || "",
+            regionCode: address?.country.toLowerCase() || "",
+          },
+        }
+  }
```

Comment:
> * `isValid`: If a user had previously saved an address with a phone number `aowiehiohdg` and a country, we would load with `isValid: true` and the user could `Save` as long as they didn't change that field.

This a good point. I guess this depends on if we want to force a user with a previously saved address to change it 🤔 

> My suggestion: call the getPhoneNumberInformation client-side as we open this form, ideally not blocking the render, and make the initial value the parsed result of that call. 

Lets try it out! Feel free to join our pairing session today. I'll send you an invite.

Reply:
good catch and recap of the problem! 

for the 2nd point (phoneNumber may actually be the international version): I believe we don't need to worry about this because `PhoneNumberInput` will turn international to national if we pass regionCode correctly.

for the 1st point: agreed with Rachel. I'm not sure if the checkout flow is the best place to force users to fix a saved phone number. let's discuss this! 

I generally like the idea of calling `getPhoneNumberInformation` on the initial render.